e2e/src/test/scala/NoBoxSpec.scala

```diff
@@ -0,0 +1,19 @@
+import org.scalatest._
+import com.trueaccord.proto.e2e.no_box.Car
+import com.trueaccord.proto.e2e.no_box.Tyre
+import org.scalatest._
+
+class NoBoxSpec extends FlatSpec with MustMatchers {
+  val car = Car(tyre1 = Tyre(size = 10), tyre2 = Some(Tyre(size = 20)))
```

Comment:
Two additional things that should be tested to future-proof this code:
1. Say a remote system written in Java sends a message that doesn't specify this optional field. Can we parse it? What are we getting?

2. I think that were going to see some inconsistency with the Java version, that is worth documenting in a test:
Say we have Car(tyre1=Tyre.defaultInstance), does the field get serialized? We need to document (by test) that if we parse these bytes in Java, the parser would give `hasTyre1==false`. At the same time, if we use Java conversions on this message, on the Java side we will have `hasTyre1==true`
(I think there's no way around it, if we fixed the conversion to not set `tyre1` in Java, it will be lossy somewhere else)

Reply:
I think it's ok for a "no_box" field to say from a Scala perspective that missing field is equivalent to default instance.
I've added specs and pushed them a few hours ago.